db/db_impl/db_impl_open.cc

```diff
@@ -582,21 +582,41 @@ Status DBImpl::Recover(
               "existing log file: ",
               file);
         } else {
-          logs.push_back(number);
+          log_files[number] =
+              LogFileName(immutable_db_options_.wal_dir, number);
         }
       }
     }
 
-    if (logs.size() > 0) {
+    if (immutable_db_options_.track_and_verify_wals_in_manifest) {
+      // Verify WALs in MANIFEST.
+      s = versions_->GetWalSet().CheckWals(env_, log_files);
+      if (!s.ok()) {
+        return Status::Corruption("Failed to verify WALs tracked in MANIFEST",
+                                  s.ToString());
+      }
+    } else if (!versions_->GetWalSet().GetWals().empty()) {
+      // Tracking is disabled, clear previously tracked WALs from MANIFEST.
```

Comment:
A little more explanation will be helpful: e.g. to prevent future recovery with track_wal=true to fail.